src/android/CHIPTool/app/src/main/res/layout/barcode_fragment.xml

```diff
@@ -1,15 +1,42 @@
-<LinearLayout xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    android:orientation="vertical"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:orientation="vertical">
 
     <com.google.chip.chiptool.setuppayloadscanner.CameraSourceView
         android:id="@+id/camera_view"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
+        android:layout_height="0dp"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintBottom_toTopOf="@id/edit_qtcode"
         android:layout_gravity="center"
         android:gravity="center"
-        android:orientation="vertical">
-    </com.google.chip.chiptool.setuppayloadscanner.CameraSourceView>
-</LinearLayout>
+        android:orientation="vertical" />
+
+    <Button
+        android:id="@+id/btn_qtcode"
```

Comment:
Change to "@+id/manualCodeBtn"?

Reply:
fixed at 8f3e4618e60c32f99046310adceb5944855ff2cc